crates/sui-core/src/authority_active/checkpoint_driver/mod.rs

```diff
@@ -566,7 +566,7 @@ where
             Ok(true)
         }
         Action::NewCert => {
-            let available_authorities: BTreeSet<_> = checkpoint
+            let available_authorities: HashSet<_> = checkpoint
```

Comment:
Are we switching to HashSet vs BTreeSet purely for efficiency? Are we sure we never iterate when we need a deterministic output?

Reply:
I switched because I had to unify two conflicting uses of Set collections and picked HashSet arbitrarily. I can change to BTreeSet if we want.

Reply:
With HashSet there is an additional burden on ensuring we do not iterate anywhere where we need determinism. If you one one day to have a deterministic testing thing, this is an additional reason to use BTreeSet.

Reply:
For deterministic testing we can also use a deterministic hash impl such as AHash, which is faster than hashbrown to boot.